### ISSUE TITLE
Symfony Flex recipes

### ISSUE THREAD

**Issue by ksn135** [symfony2admingenerator/GeneratorBundle#310]
Hi there!

Do someone have any plans to provide [Flex](https://github.com/symfony/flex) recipe for this bundle ?
May be not in ["official"](https://github.com/symfony/recipes-contrib) BUT in ["contrib"](https://github.com/symfony/recipes-contrib) repository

PS: I'm again/still full of plans, projects and deadlines. 
Planing to start a new big project based on s2a next week.
What is the current state of this The Great Bundle (no jokes, all caps)? 
The latest tag (v2.1.4) is targeting to what Symfony version?
Are we compatible with 3.3-DEV ? Any pitfalls ? May be I could help?
Thanks for answers in advance.

/cc: @loostro @sescandell @bobvandevijver

**Comment by sescandell** [comment]
Hi @ksn135 

I think we should provide a recipe for Flex. 

If I'm not wrong, we've made some updates to support Symfony 2.8+ including 3.x+ (check the [composer file](https://github.com/symfony2admingenerator/GeneratorBundle/blob/master/composer.json#L24) ). But, AFAIK, we never updated it to the latests functionnalities available in Symfony (Cache and DI improvements for example). So, the bundle IS compatible with 3.3 but doesn't use its latest capacities.

**Comment by TheCelavi** [comment]
I was just playing around with SF 3.4 / SF 4 and integration of this bundle with it.

Not good - new directory structure is real a game changer, seams that major refactoring is going to be required...

**Comment by ksn135** [comment]
More than that, there is NO MORE sensio/generator-bundle.
"It does not support Symfony 4. It also does not support the new bundle-less directory structure as created by Symfony Flex. Use the Maker bundle instead."
BUT in symfony/maker-bundle there is still NO CRUD!!! Yet.
https://github.com/symfony/maker-bundle/issues/3 
Do we plan to start "major refactoring" ? Do we have enough resources?
Or just leave it (s2a) alone and let it disappear in the depths of the times? 
RIP ? @sescandell @loostro @tobias-93 @TheCelavi @bobvandevijver 
Let's discuss it.

**Comment by bobvandevijver** [comment]
Yeah, I've read about the missing generator-bundle, but I do not think that will be the biggest problem as it is actually not used [**that** much](https://github.com/symfony2admingenerator/GeneratorBundle/search?utf8=%E2%9C%93&q=Sensio%5CBundle%5CGeneratorBundle&type=). 

I see more challenges with the new directory structure, which will require us to do major refactoring throughout almost every part of the bundle. However, we do use this bundle in a big project, but that will never move to the new flex structure (as the flex structure is not (yet) required). 

So that raises the question: should we support both structures? Should we split a new flex-branch?

About the resources: I doubt we have any real resources at the moment... For example, I simply do not have the time to start which such a major refactoring... And there are other alternatives, such as [EasyAdmin](https://github.com/EasyCorp/EasyAdminBundle), which is actually also featured on the Symfony site.

So we have to wonder (also because we currently do not have a large count of very active users as far as I know), is it worth it?

**Comment by TheCelavi** [comment]
I am not for throwing towel, yet. 

> Should we split a new flex-branch?

I believe that opportunity must be taken to fix/improve all the things that we have stumbled upon, starting from generating admin for entities with same name under different namespace, disabled controller actions by default... per example, we have managed to implement webpack (encore) with this bundle's assets - why not make it default? 

I would also recommend to revise features, what to implement in core, what should be sub-split in separate plug-in bundles.

> And there are other alternatives, such as EasyAdmin

True. However, this bundle has been doing same job for longer time, and it does it much better. Much better - in terms of possibilities for customization of literally everything. Everyone can generate admin for entities, issues are raised when that edge case kicks in - and GeneratorBundle handles those cases marvellously.

> So we have to wonder (also because we currently do not have a large count of very active users as far as I know), is it worth it?

How much excellent developers are needed to handle one project? 

We are here, we have active projects which depends on this bundle, we intend to use it for future projects as well - tell us what to do and how to help and we will.

**Comment by ksn135** [comment]
I agree with @TheCelavi: "tell us what to do and how to help and we will."
Probably we should change naming, may be ```s4a``` prefix ?

**Comment by loostro** [comment]
Hello guys, I'm afraid I won't be able to contribute. For some time now I've been working with Scala and have no active PHP projects, while my private time is mostly occupied by my kids.

If I can offer some advice if I were you I'd take the opportunity to completely rewrite the bundle and revise it's offered features.

Cheers
